### PR TITLE
Fix pg array parsing of unquoted string witn many whitespaces

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Follow up to https://github.com/crate/crate/pull/11027 and https://github.com/crate/crate/pull/11051

### CODE REVIEW

**Comment by BaurzhanSakhariev** [comment]
Hi @matriv, could you please take look?
First commit lead to test failures like ```extraneous input '"ab"' expecting... ```, for example in `test_string_array` as I leading whitespaces started to be treated as unquoted string instead of being skipped.

In the [fixup](https://github.com/crate/crate/pull/13527/commits/d921fe4052dfa30d8fbcd88777a1e8b01b733374) whitespaces kept as a non-char to let WS rule do its job and _multi words with at least one word_ is expressed similar to array items (`item (',' item)*`)

File: server/src/test/java/io/crate/protocols/postgres/types/PGArrayTest.java [hunk 1]
```diff
@@ -133,6 +133,26 @@ public void test_can_decode_unquoted_strings_which_do_not_contain_numbers() thro
         ));
     }
 
+    @Test
+    public void test_can_decode_unquoted_words_with_multiple_spaces() throws Exception {
+        String s = "{ a b c }";
+        List<Object> values = PGArray.VARCHAR_ARRAY.decodeUTF8Text(s.getBytes(StandardCharsets.UTF_8));
+        // leading/trainling whitespaces are ignored in the unquotes strings
```
Comment (matriv) [inline server/src/test/java/io/crate/protocols/postgres/types/PGArrayTest.java:140]:
```suggestion
        // leading/trailing whitespaces are ignored in the unquotes strings
```

**Review by matriv — APPROVED** [review]
LGTM. Thank you!
Could you also please make the 2 tests a bit more complex to check multiple spaces, tabs, `"` char, etc? e.g.:
```
@Test
public void test_lala() throws Exception {
    String s = "{\" a b c\\t \",\"foo \\\" bar\",\"My     name              is Marios\"}";
    List<Object> values = PGArray.VARCHAR_ARRAY.decodeUTF8Text(s.getBytes(StandardCharsets.UTF_8));
    assertThat(values, contains(
        " a b c\\t ", "foo \" bar", "My     name              is Marios"));
}
```

File: server/src/test/java/io/crate/protocols/postgres/types/PGArrayTest.java [hunk 2]
```diff
@@ -133,6 +133,28 @@ public void test_can_decode_unquoted_strings_which_do_not_contain_numbers() thro
         ));
     }
 
+    @Test
+    public void test_can_decode_unquoted_words_with_multiple_spaces() throws Exception {
+        String s = "{ a b c , d e f }";
```
Comment (matriv) [inline server/src/test/java/io/crate/protocols/postgres/types/PGArrayTest.java:138]: If you think it's not too much maybe something like this:
```
String s = "{ \ta   b c  , \t d e   f  \t }";
```

```
assertThat(values, contains(
    "a   b c",
    "d e   f"
));
```

**Review by matriv — APPROVED** [review]
LGTM, left one more suggestion., Thx!

File: server/src/test/java/io/crate/protocols/postgres/types/PGArrayTest.java (hunk 2 above)
Comment (BaurzhanSakhariev) [reply server/src/test/java/io/crate/protocols/postgres/types/PGArrayTest.java]: Good point about tabs - I forgot that we can use them in the beginning of the unquoted tokens due to WS skip rule. Thanks!

Comment (matriv) [reply server/src/test/java/io/crate/protocols/postgres/types/PGArrayTest.java]: thx!

**Comment by BaurzhanSakhariev** [comment]
retest this please

**Comment by BaurzhanSakhariev** [comment]
Hi @matriv, could you please re-review, looks like status became stale after pushing test changes

**Pull request merged**